### DISCUSSION TITLE
GPG Sign in GitHub Actions

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [commitizen-tools/commitizen#616] [Q&A]
I've submitted [PR #51](https://github.com/commitizen-tools/commitizen-action/pull/51) to `commitizen-tools/commitizen-action`, but cannot seem to get GPG signing to work.

I consistently hit the error:

```
2nd git.commit error: "error: gpg failed to sign the data fatal: failed to write commit object"
```

which is coming from `commands/bump.py`:

https://github.com/commitizen-tools/commitizen/blob/8dcfac08593d1fb0af7ab0f437d1bd8eb125923b/commitizen/commands/bump.py#L262-L264

If I'm reading the source correctly, it looks like `bump` supports signing tags, but not commits. Is this right?

@woile @Lee-W Can you please confirm?

**Comment by adam-grant-hendry** [comment]
@woile @Lee-W I see what's happening. `commitizen-action` runs in a separate Docker container, whereas the action `crazy-max/ghaction-import-gpg` that I use to set up `GPG` runs in the local runner.

Hence, `crazy-max/ghaction-import-gpg` sets up the local runner, but `commitizen-action` builds and runs a container, so none of the configuration from `crazy-max/ghaction-import-gpg` is there.

For now, it's probably simplest for me to copy-paste the action [`entrypoint.sh`](https://github.com/commitizen-tools/commitizen-action/blob/master/entrypoint.sh) script and run that directly in my workflow.

Not sure how to run a GitHub Action within a workflow step that builds and launches a Docker container like `commitizen-action` does...?

**Reply by Lee-W** [reply]
I notice you opened an issue on `commitizen-action`. Thus, I'll close this one and that's discuss there. I'm not that familiar with github-action development as of now. But will get back to you if i have some time to dig into it.
